src/arraystring.rs

```diff
@@ -712,10 +726,29 @@ impl<const N: usize> ArrayString<N> {
     #[inline]
     pub fn try_push(&mut self, character: char) -> Result<(), OutOfBounds> {
         trace!("Push: {}", character);
-        let new_end = character.len_utf8().saturating_add(self.len().into());
-        is_inside_boundary(new_end, Self::capacity())?;
-        unsafe { self.push_unchecked(character) };
-        Ok(())
+        let last_size = self.len();
+        self.push(character);
+        (self.size != last_size).then_some(()).ok_or(OutOfBounds)
+    }
+
+    /// Inserts character to the end of the `ArrayString` truncating if no room is left
+    ///
+    /// ```rust
+    /// # use arraystring::{error::Error, prelude::*};
+    /// # fn main() -> Result<(), Error> {
+    /// let mut s = SmallString::try_from_str("My String")?;
+    /// s.push('!');
+    /// assert_eq!(s.as_str(), "My String!");
+    ///
+    /// // s = SmallString::try_from_str(&"0".repeat(SmallString::capacity().into()))?;
+    /// // Undefined behavior, don't do it
+    /// // s.push('!');
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn push(&mut self, ch: char) {
+        self.size = self.size.saturating_add(append_char_truncate(self.remaining_mut(), ch).into_lossy());
```

Comment:
Won't this break with ArrayString<255> with 255 as the length?

Reply:
This is a checked version of the unchecked, it should be as performant as the unchecked one, but truncates automatically. We should add a bench to formalize that.